php-checks/src/main/java/org/sonar/php/checks/TooManyFieldsInClassCheck.java

```diff
@@ -122,4 +128,24 @@ private static boolean isNonPublic(List<SyntaxToken> modifiers) {
     return false;
   }
 
+  private int getNumberOfConstructorProperties(ClassTree tree) {
+    var constructors = getConstructors(tree);
+    return (int) constructors.map(constructor -> constructor.parameters().parameters())
+      .flatMap(Collection::stream)
+      .filter(ParameterTree::isPropertyPromotion)
+      .filter((ParameterTree parameter) -> {
+        if (!countNonpublicFields) {
+          return "public".equals(parameter.visibility().text());
```

Comment:
```suggestion
          return "public".equalsIgnoreCase(parameter.visibility().text());
```
In PHP, keywords among other things are case insensitive.